webapp/cas/cmd/cas-test/main.go

```diff
@@ -89,24 +95,33 @@ func getUserData(path string) (string, error) {
 }
 
 func injectUser(page *agouti.Page, userData string) error {
-	if err := page.Navigate(omniledgerURL); err != nil {
+	if err := page.Navigate(injectUserURL); err != nil {
 		return err
 	}
 
 	storageDBInjector := `
-	const dynasentDone = new Promise((resolve) =>
-		window.indexedDB.open("dynasent", 10).onupgradeneeded = (event) =>
+	const dynasentDone = new Promise((resolve) => {
+		const request = window.indexedDB.open("dynasent", 10)
+		request.onupgradeneeded = (event) =>
 			event.target.result.
 				createObjectStore("contacts", {keyPath: "key"}).
-				put(JSON.parse(data)).onsuccess = resolve)
+				put(JSON.parse(data)).onsuccess
+		request.onsuccess = resolve
+	})
 
-	const dbnamesDone = new Promise((resolve) =>
-		window.indexedDB.open("__dbnames", 10).onupgradeneeded = (event) => {
-			const store = event.target.result.
-				createObjectStore("dbnames", {keyPath: "name"})
-			store.createIndex("buffer", "buffer", {unique: false, multiEntry: false})
-			store.put({name: "dynasent"}).onsuccess = resolve
-		})
+	const dbnamesDone = new Promise((resolve) => {
+		const objectStoreName = "dbnames"
+		const request = window.indexedDB.open("__dbnames", 10)
+		request.onupgradeneeded = (event) =>
+			event.target.result.
+				createObjectStore(objectStoreName, {keyPath: "name"}).
+				createIndex("buffer", "buffer", {unique: false, multiEntry: false})
+		request.onsuccess = (event) =>
+			event.target.result.transaction(objectStoreName, "readwrite").
+				objectStore(objectStoreName).
+				put({name: "dynasent"}).
+				onsuccess = resolve
+	})
 
 	Promise.all([dynasentDone, dbnamesDone]).then(() => {
```

Comment:
Is it possible to bubble up an error to go if you add `catch` here?

Reply:
I don't think there is with `github.com/sclevine/agouti.Page.RunScript`: as all the processing is done in async, the script actually returns `undefined` and I use this `document.createElement` trick to actually wait for the end.
Hum, but I can actually return a potential error message _inside_ the `span`, that's a nice touch :)